### PR TITLE
GUVNOR-3466: [Guided Decision Table] Same Data Object can be imported multiple times

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/GUVNOR-3466

This PR filters imports that have already been imported; and adds sorting to the various lists.

I seem to recall there's a JIRA for sorting.. but I cannot find it.

### CODE REVIEW

File: kie-wb-common-widgets/kie-wb-config-resource-widget/src/test/java/org/kie/workbench/common/widgets/configresource/client/widget/bound/ImportsWidgetPresenterTest.java [hunk 1]
```diff
@@ -158,9 +191,6 @@ public void isInternalImportInternalImportType() {
 
     @Test
     public void isInternalImportExternalImportType() {
```
Comment (jomarko) [inline kie-wb-common-widgets/kie-wb-config-resource-widget/src/test/java/org/kie/workbench/common/widgets/configresource/client/widget/bound/ImportsWidgetPresenterTest.java]: This method contains assertion:
```
assertFalse(presenter.isInternalImport(mock(Import.class)));
```
Shouldn't be this?
```
assertFalse(presenter.isInternalImport(importType));
```

File: kie-wb-common-widgets/kie-wb-config-resource-widget/src/main/java/org/kie/workbench/common/widgets/configresource/client/widget/bound/AddImportPopup.java [hunk 2]
```diff
@@ -42,7 +43,9 @@
 
     }
 
-    private static AddGlobalPopupBinder uiBinder = GWT.create( AddGlobalPopupBinder.class );
+    private static final Comparator<Import> IMPORT_COMPARATOR = (o1, o2) -> o1.getType().compareTo(o2.getType());
```
[the remark this answers is not in the document]
Comment (triceo) [reply kie-wb-common-widgets/kie-wb-config-resource-widget/src/main/java/org/kie/workbench/common/widgets/configresource/client/widget/bound/AddImportPopup.java]: Also, Java 8 has this [very nice fluent API](https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html) for `Comparator`, that lets you do fancy things like: `Comparator.comparing(Import::getType).thenComparing(...).nullsFirst();`.

[the remark this answers is not in the document]
Comment (manstis) [reply kie-wb-common-widgets/kie-wb-config-resource-widget/src/main/java/org/kie/workbench/common/widgets/configresource/client/widget/bound/AddImportPopup.java]: @jomarko The ```comparator``` in ```AddImportPopup``` only orders by FCQN whereas the one in ```ImportsWidgetViewImpl``` sorts first by whether the class is "internal" (i.e. is within the package) and then by FQCN. I've made a change to maximise re-use and taken @triceo advice.

File: kie-wb-common-widgets/kie-wb-config-resource-widget/src/main/java/org/kie/workbench/common/widgets/configresource/client/widget/bound/ImportsWidgetViewImpl.java [hunk 3]
```diff
@@ -115,7 +139,7 @@ public void render(final com.google.gwt.cell.client.Cell.Context context,
                                final SafeHtml data,
                                final SafeHtmlBuilder sb) {
                 //Don't render a "Delete" button for "internal" Fact Types
-                if (!presenter.isInternalImport(dataProvider.getList().get(context.getIndex()))) {
+                if (!presenter.isInternalImport(getDataProvider().getList().get(context.getIndex()))) {
```
[the remark this answers is not in the document]
Comment (manstis) [reply kie-wb-common-widgets/kie-wb-config-resource-widget/src/main/java/org/kie/workbench/common/widgets/configresource/client/widget/bound/ImportsWidgetViewImpl.java]: Done (as best I see possible)

File: kie-wb-common-widgets/kie-wb-config-resource-widget/src/test/java/org/kie/workbench/common/widgets/configresource/client/widget/bound/ImportsWidgetPresenterTest.java (hunk 1 above)
Comment (manstis) [reply kie-wb-common-widgets/kie-wb-config-resource-widget/src/test/java/org/kie/workbench/common/widgets/configresource/client/widget/bound/ImportsWidgetPresenterTest.java]: Yes. Well spotted!

**Comment by jomarko** [comment]
@manstis please notify me when you push changes.

**Comment by manstis** [comment]
@jomarko All changes made. I also took this opportunity (as you wanted to remove ```Window.confirm(..)``` to refactor the related widgets etc that are used to specify imports in the Project Settings->External Data Objects too (as that also used ```Window.confirm(..)```.

**Pull request merged**